File: app/src/main/kotlin/ca/etsmtl/etsmobile/data/model/SignetsModel.java
package ca.etsmtl.etsmobile.data.model;

import com.squareup.moshi.Json;

/**
 * Created by Sonphil on 09-03-18.
 */

public class SignetsModel<T> {
    @Json(name = "d")
    private T data;

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: app/src/main/kotlin/ca/etsmtl/etsmobile/data/model/SignetsModel.kt
package ca.etsmtl.etsmobile.data.model

import com.squareup.moshi.Json

/**
 * Created by Sonphil on 09-03-18.
 */

class SignetsModel<T : SignetsData> {
    @Json(name = "d")
    var data: T? = null
}
